services/venice-server/src/main/java/com/linkedin/venice/listener/StatsHandler.java

```diff
@@ -204,15 +205,15 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
             : serverStatsContext.getCurrentStats().getStoreStats(serverStatsContext.getStoreName());
         serverStatsContext.recordBasicMetrics(serverHttpRequestStats);
         double elapsedTime = LatencyUtils.getLatencyInMS(serverStatsContext.getRequestStartTimeInNS());
-        // if ResponseStatus is either OK or NOT_FOUND and the channel write is succeed,
+        // if ResponseStatus is either OK or NOT_FOUND or TOO_MANY_REQUESTS and the channel write is succeed,
         // records a successRequest in stats. Otherwise, records a errorRequest in stats;
         if (result.isSuccess() && (serverStatsContext.getResponseStatus().equals(OK)
-            || serverStatsContext.getResponseStatus().equals(NOT_FOUND))) {
+            || serverStatsContext.getResponseStatus().equals(NOT_FOUND)
+            || serverStatsContext.getResponseStatus().equals(TOO_MANY_REQUESTS))) {
```

Comment:
I'm wondering if we should call `successRequest` for `TOO_MANY_REQUESTS`. In today's router we still emit too many requests as unhealthy: https://github.com/linkedin/venice/blob/fbcace8e57302661a81ca85f698e370a216f134b/services/venice-router/src/main/java/com/linkedin/venice/router/api/RouterExceptionAndTrackingUtils.java#L142

I understand that we shouldn't call `errorRequest` since `TOO_MANY_REQUESTS` is an user error and shouldn't trigger the same alert that makes Venice devs believe something is wrong with the server. However, I'm concerned that calling `successRequest` on `TOO_MANY_REQUESTS` may unintentionally obfuscate our success metric. Perhaps we should simply do nothing.

Reply:
As discussed offline, 429 is not counted against either success or error requests.